src/VisualStudio/Core/Def/Implementation/MoveStaticMembers/MoveStaticMembersDialogViewModel.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.MoveStaticMembers
+{
+    internal class MoveStaticMembersDialogViewModel : AbstractNotifyPropertyChanged
+    {
+        public StaticMemberSelectionViewModel MemberSelectionViewModel { get; }
+
+        private readonly ISyntaxFacts _syntaxFacts;
+
+        private readonly ImmutableArray<string> _existingNames;
+
+        public MoveStaticMembersDialogViewModel(
+            StaticMemberSelectionViewModel memberSelectionViewModel,
+            string defaultType,
+            ImmutableArray<string> existingNames,
+            ISyntaxFacts syntaxFacts)
+        {
+            MemberSelectionViewModel = memberSelectionViewModel;
+            _syntaxFacts = syntaxFacts ?? throw new ArgumentNullException(nameof(syntaxFacts));
+            _destinationName = defaultType;
+            _existingNames = existingNames;
+
+            PropertyChanged += MoveMembersToTypeDialogViewModel_PropertyChanged;
+            OnDestinationUpdated();
+        }
+
+        private void MoveMembersToTypeDialogViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(DestinationName):
+                    OnDestinationUpdated();
+                    break;
+            }
+        }
+
+        public void OnDestinationUpdated()
+        {
+            // TODO change once we allow movement to existing types
+            var isNewType = !_existingNames.Contains(DestinationName);
+            _isValidName = IsValidType(DestinationName);
+
+            if (isNewType && _isValidName)
+            {
+                Icon = KnownMonikers.StatusInformation;
+                Message = "[WIP] A new type will be created";
```

Comment:
Add resources for these strings as well (`A_new_type_will_be_created` and `Invalid_type_name`)